frontend/demo/component/treegrid/treegrid-rich-content.ts

```diff
@@ -0,0 +1,127 @@
+import '../../init'; // hidden-full-source-line
+import '@vaadin/flow-frontend/gridConnector.js'; // hidden-full-source-line (Grid's connector)
+
+import { customElement, LitElement, query } from 'lit-element';
+import { html } from 'lit-html';
+import '@vaadin/vaadin-grid/vaadin-grid';
+import '@vaadin/vaadin-grid/vaadin-grid-tree-toggle';
+import '@vaadin/vaadin-ordered-layout/vaadin-horizontal-layout';
+import '@vaadin/vaadin-ordered-layout/vaadin-vertical-layout';
+import '@vaadin/vaadin-icons/vaadin-icons';
+import '@vaadin/vaadin-button/vaadin-button';
+import {
+  GridColumnElement,
+  GridDataProviderCallback,
+  GridDataProviderParams,
+  GridElement,
+  GridItemModel
+} from '@vaadin/vaadin-grid/vaadin-grid';
+import { getPeople } from '../../domain/DataService';
+import Person from '../../../generated/com/vaadin/demo/domain/Person';
+import { applyTheme } from 'generated/theme';
+import { GridTreeToggleExpandedChanged } from '@vaadin/vaadin-grid/vaadin-grid-tree-toggle';
+import { render } from 'lit-html';
+
+@customElement('treegrid-rich-content')
+export class Example extends LitElement {
+  constructor() {
+    super();
+    // Apply custom theme (only supported if your app uses one)
+    applyTheme(this.shadowRoot);
+  }
+
+  @query('vaadin-grid')
+  private grid?: GridElement;
+
+  async dataProvider(params: GridDataProviderParams, callback: GridDataProviderCallback) {
+    let people: Person[];
+
+    if (params.parentItem) {
+      const manager = params.parentItem as Person;
+      people = await getPeople({ managerId: manager.id });
+    } else {
+      people = await getPeople({ managerId: null });
+    }
+
+    const startIndex = params.page * params.pageSize;
+    const pageItems = people.slice(startIndex, startIndex + params.pageSize);
+    // Inform grid of the requested tree level's full size
+    const treeLevelSize = people.length;
+    callback(pageItems, treeLevelSize);
+  }
+
+  // tag::snippet[]
+  employeeRenderer(root: HTMLElement, _column?: GridColumnElement, model?: GridItemModel) {
+    const { grid } = this;
+
+    if (model?.item) {
+      const person = model.item as Person;
+
+      render(
+        html`
+          <vaadin-horizontal-layout>
+            <vaadin-grid-tree-toggle
+              .leaf=${!person.hasChildren}
+              .level=${model.level!}
+              @expanded-changed=${(e: GridTreeToggleExpandedChanged) => {
+                grid && grid[e.detail.value ? 'expandItem' : 'collapseItem'](person);
+              }}
+              .expanded=${model.expanded!}
```

Comment:
Using non-null assertion operator is discouraged (by typescript-eslint)
```suggestion
              .expanded=${!!model.expanded}
```

Reply:
Changed.